### PR TITLE
Clarify some statements in the owned entity types docs

### LINKED ISSUE #1239: Please clarify "inheritance hierarchy" limitation
This article states:

&gt; **Current shortcomings**
&gt; Inheritance hierarchies that include owned entity types are not supported

I interpreted this to mean that entities that are in an inheritance hierarchy cannot include owned entities.  But, according to [this GitHub discussion](https://github.com/aspnet/EntityFrameworkCore/issues/14451#issuecomment-455362225) the limitation means that the owned entities themselves cannot be part of an inheritance hierarchy.  Please clarify this detail.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d02e6748-42e0-000a-4945-42cd05a0b875
* Version Independent ID: e3effbd0-07f2-423d-4032-1722e75adb37
* Content: [Owned Entity Types - EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities)
* Content Source: [entity-framework/core/modeling/owned-entities.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/modeling/owned-entities.md)
* Product: **entity-framework**
* GitHub Login: @AndriySvyryd
* Microsoft Alias: **ansvyryd**

### LINKED ISSUE #1400: Add more details on sharing the same .NET  owned type among multiple owning types.
The immediate question which comes to my mind is whether it is possible to reuse owned type in various owing types, configuring table mappings and other stuff per owning type.

For example: 

`Organisation` owns `Address` which is mapped to `OrgAddress` table, whereas `User` owns `Address` mapped to `UserAddress` table. Is it ok to have single type `Address` or I need two different types for that?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d02e6748-42e0-000a-4945-42cd05a0b875
* Version Independent ID: e3effbd0-07f2-423d-4032-1722e75adb37
* Content: [Owned Entity Types - EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities#sharing-the-same-net-type-among-multiple-owned-types)
* Content Source: [entity-framework/core/modeling/owned-entities.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/modeling/owned-entities.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @AndriySvyryd
* Microsoft Alias: **ansvyryd**

### LINKED ISSUE #1457: Ignoring properties in an owned type using fluent API
How should one ignore properties in an owned type using fluent call? I have figured out how to do this with attributes by adding `[NotMapped]` to the property, but not how the same is done using Fluent API.

In EF 6 I was able to ignore properties on a complex type like this:
`modelBuilder.ComplexType<StreetAddress>().Ignore(a => a.City);`

In EF Core 2.2, it is possible to define the type but not continue with Ignore:
`modelBuilder.Owned<StreetAddress>();`

The by-design restrictions tell me I cannot do this:
`modelBuilder.Entity<StreetAddress>().Ignore(a => a.City);`

I think it would be helpful with a section in the [documentation of Owned Entity Types](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities) about ignoring properties in an owned type using both attributes and fluent API.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d02e6748-42e0-000a-4945-42cd05a0b875
* Version Independent ID: e3effbd0-07f2-423d-4032-1722e75adb37
* Content: [Owned Entity Types - EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities)
* Content Source: [entity-framework/core/modeling/owned-entities.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/modeling/owned-entities.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @AndriySvyryd
* Microsoft Alias: **ansvyryd**

### LINKED ISSUE #1628: Add explanation and examples for conventions related to Owned -> Owner Navigational Properties
There seems to be a convention that a property matching the type of the Owner entity on the Owned entity will be automatically used as a navigation back to the owner.

This isn't documented, and presents challenges when trying to work around this.

**Example:**

**Note:** This is a crappy example that doesn't really convey a real-life use case, but does highlight the confusion from the conventions being unknown...

```cs
    public class Person
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public RelatedPerson Related { get; set; }
    }

    [Owned]
    public class RelatedPerson
    {
        // Is this now a navigation property back to the owner, or is this a distinct Person separate from the owner?
        // What if this is supposed to not be a navigation property? How is that configured?
        public Person Person { get; set; }

        // If Person is a navigational property to the owner, is this Id set appropriately? If not?
        public int PersonId { get; set; }
    }
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d02e6748-42e0-000a-4945-42cd05a0b875
* Version Independent ID: e3effbd0-07f2-423d-4032-1722e75adb37
* Content: [Owned Entity Types - EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities#feedback)
* Content Source: [entity-framework/core/modeling/owned-entities.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/modeling/owned-entities.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @AndriySvyryd
* Microsoft Alias: **ansvyryd**

### LINKED ISSUE #1629: Add more conceptual documentation around aggregates
The documentation makes the following claim about references from owned types:

> In addition to nested owned types, an owned type can reference a regular entity, it can be either the owner or a different entity as long as the owned entity is on the dependent side.

It is not made clear **why** the owned entity must be on the dependent side of the relationship. Is this limitation by-design or a current shortcoming?

A concrete example of a scenario that would benefit from this capability was outlined in the stackoverflow question:
[https://stackoverflow.com/q/54647000/2196664](https://stackoverflow.com/q/54647000/2196664)

When talking about DDD, having value objects reference a collection of entities does not seem to be a restriction:

> VALUE OBJECTS can even reference ENTITIES. For example, if I ask an online map service for a
scenic driving route from San Francisco to Los Angeles, it might derive a Route object linking L.A.
and San Francisco via the Pacific Coast Highway. That Route object would be a VALUE, even though
the three objects it references (two cities and a highway) are all ENTITIES.
> _-- Eric Evans, Domain-Driven Design: Tackling Complexity in the Heart of Software, page 98_

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d02e6748-42e0-000a-4945-42cd05a0b875
* Version Independent ID: e3effbd0-07f2-423d-4032-1722e75adb37
* Content: [Owned Entity Types - EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities#feedback)
* Content Source: [entity-framework/core/modeling/owned-entities.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/modeling/owned-entities.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @AndriySvyryd
* Microsoft Alias: **ansvyryd**

### LINKED ISSUE #1781: Misleading statement about collections of owned types regarding the default keys
The documentation states:

> The two most straightforward solutions to this are: 
> ... Using the foreign key and an additional property as a composite key. 
> ... This is what EF Core does by default.

> By default the primary key used for the owned type referenced through the ShippingCenters navigation property will be ("DistributorId", "Id") where "DistributorId" is the FK and "Id" is a unique int value.

I tried copy-pasting the given example regarding the Street Address and the Distributor but it seems unable to create the relevant migration unless you explicitly define the shadowed key and the composite foreign key...

```    
[Owned]
public class StreetAddress
{
    public string Street { get; set; }
    public string City { get; set; }
}
```

```
public class Distributor
{
    public int Id { get; set; }
    public ICollection&lt;StreetAddress&gt; ShippingCenters { get; set; }
}
```

```        
protected override void OnModelCreating(ModelBuilder modelBuilder)
{
    modelBuilder.Entity&lt;Distributor&gt;().OwnsMany(p => p.ShippingCenters);
}
```

```
System.InvalidOperationException: The entity type 'StreetAddress' requires a primary key to be defined.
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelValidator.ValidateNonNullPrimaryKeys(IModel model)
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelValidator.Validate(IModel model)
   at Microsoft.EntityFrameworkCore.Infrastructure.RelationalModelValidator.Validate(IModel model)
   at Microsoft.EntityFrameworkCore.Internal.SqlServerModelValidator.Validate(IModel model)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ValidatingConvention.Apply(InternalModelBuilder modelBuilder)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ImmediateConventionScope.OnModelBuilt(InternalModelBuilder modelBuilder)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.OnModelBuilt(InternalModelBuilder modelBuilder)
   at Microsoft.EntityFrameworkCore.Metadata.Internal.Model.Validate()
   at Microsoft.EntityFrameworkCore.ModelBuilder.FinalizeModel()
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.CreateModel(DbContext context, IConventionSetBuilder conventionSetBuilder, IModelValidator validator)
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.&lt;&gt;c__DisplayClass5_0.&lt;GetModel&gt;b__1()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.GetModel(DbContext context, IConventionSetBuilder conventionSetBuilder, IModelValidator validator)
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.CreateModel()
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.get_Model()
   at Microsoft.EntityFrameworkCore.Infrastructure.EntityFrameworkServicesBuilder.&lt;&gt;c.&lt;TryAddCoreServices&gt;b__7_2(IServiceProvider p)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScoped(ScopedCallSite scopedCallSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.&lt;&gt;c__DisplayClass1_0.&lt;RealizeService&gt;b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
   at Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
   at Microsoft.EntityFrameworkCore.DbContext.Microsoft.EntityFrameworkCore.Infrastructure.IInfrastructure&lt;System.IServiceProvider&gt;.get_Instance()
   at Microsoft.EntityFrameworkCore.Internal.InternalAccessorExtensions.GetService[TService](IInfrastructure`1 accessor)
   at Microsoft.EntityFrameworkCore.Infrastructure.AccessorExtensions.GetService[TService](IInfrastructure`1 accessor)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(Func`1 factory)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(String contextType)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.AddMigration(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigration.&lt;&gt;c__DisplayClass0_1.&lt;.ctor&gt;b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.&lt;&gt;c__DisplayClass3_0`1.&lt;Execute&gt;b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
The entity type 'StreetAddress' requires a primary key to be defined.
PM&gt;
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d02e6748-42e0-000a-4945-42cd05a0b875
* Version Independent ID: e3effbd0-07f2-423d-4032-1722e75adb37
* Content: [Owned Entity Types - EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities#feedback)
* Content Source: [entity-framework/core/modeling/owned-entities.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/modeling/owned-entities.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @AndriySvyryd
* Microsoft Alias: **ansvyryd**

### LINKED ISSUE #1888: Separate Tables by attribute
The page leaves out documenting or mentioning if not possible how to store owned types in separate tables using attributes instead of Fluent Configuration.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d02e6748-42e0-000a-4945-42cd05a0b875
* Version Independent ID: e3effbd0-07f2-423d-4032-1722e75adb37
* Content: [Owned Entity Types - EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities#storing-owned-types-in-separate-tables)
* Content Source: [entity-framework/core/modeling/owned-entities.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/modeling/owned-entities.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @AndriySvyryd
* Microsoft Alias: **ansvyryd**

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #1239
Fixes #1400
Fixes #1457
Fixes #1628
Fixes #1629
Fixes #1781
Fixes #1888

### CODE REVIEW

File: entity-framework/core/modeling/owned-entities.md [hunk 1]
```diff
@@ -100,6 +103,8 @@ In this example `OrderDetails` owns `BillingAddress` and `ShippingAddress`, whic
 
 [!code-csharp[OrderStatus](../../../samples/core/Modeling/OwnedEntities/OrderStatus.cs?name=OrderStatus)]
 
+Each navigation to an owned type defined a separate entity type with completely independent configuration.
```
Comment (ajcvickers) [inline entity-framework/core/modeling/owned-entities.md]: defines?

**Comment by ajcvickers** [comment]
Should we explicitly call out that owned types are not a good replacement for complex types and do not closely match the semantics for DDD-style value objects?

**Comment by AndriySvyryd** [comment]
@ajcvickers I think we need to provide a better alternative first (multi-column value converters) for the warning to be effective.

**Comment by voroninp** [comment]
@ajcvickers  And if you make this remark, then, please,explain, why it is not a good fit for DDD ;-)

I am also eagerly watinig for the feature to setup aggregate shape, so Includes are called automatically. That's actually the main reason why I use owned types.

**Pull request merged**